sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -257,6 +257,75 @@ class StreamingInnerJoinSuite extends StreamingJoinSuite {
     )
   }
 
+
```

Comment:
Lets also add a testcase for join condition where we compare eventTime and some other attribute (example - id)

See https://github.com/apache/spark/pull/44323/files#r1582300122 for context.